clippy_config/src/msrvs.rs

```diff
@@ -26,7 +26,7 @@ msrv_aliases! {
     1,65,0 { LET_ELSE, POINTER_CAST_CONSTNESS }
     1,63,0 { CLONE_INTO }
     1,62,0 { BOOL_THEN_SOME, DEFAULT_ENUM_ATTRIBUTE, CONST_EXTERN_FN }
-    1,59,0 { THREAD_LOCAL_INITIALIZER_CAN_BE_MADE_CONST }
+    1,59,0 { MISSING_CONST_FOR_THREAD_LOCAL }
```

Comment:
The MSRV entry should describe the feature we're using rather than a lints name that uses it, borrowing the old feature name is pretty common https://github.com/rust-lang/rust/pull/91355

Reply:
The old name being the lint name wasn't great either, while we're here let's rename it to `THREAD_LOCAL_CONST_INIT`

Reply:
ok